workshop_R.Rmd

```diff
@@ -0,0 +1,779 @@
+
+```
+student_results_aggregated_suppressed <- student_results_aggregated %>%
+  mutate(g1_mean = ifelse(students < 5, 'c', g1_mean),
+         g2_mean = ifelse(students < 5, 'c', g2_mean),
+         g3_mean = ifelse(students < 5, 'c', g3_mean),
+         students = ifelse(students < 5, 'c', students)
+         )
+```
+
+The above code takes the mean grade columns and uses an `ifelse()` statement *within* a `mutate()` statement. 
+Here, it's saying for each of the mean columns, if the number of students is less than 5, then replace the entry in that mean grade column with a 'c', if not, then keep what is already in the column. Note that we *have** to do the students column last, because once we add 'c' to any column it becomes a character variable rather than numeric. Since the previous columns are based on the numeric value of the students, we must order it so that these mutations occur first, while the student column is still a numeric variable and therefore able to apply the `students < 5` logic. 
+
+While the above code works, the same issues arise as in the previous aggregation section where you can end up copying and pasting the same line many times depending on the size of your real-life data sets! We want to avoid this type of repetition where possible so that we reduce the possibility of making mistakes (like forgetting to update every line you have copied and pasted). So let's create a function to suppress counts less than 5. With an appropriate comment to explain, ass the below to your functions.R script. 
+
+```
+suppress_counts <- function(column, count) {
+  ifelse(count < 5, 'c', column)
+}
+```
+
+The above function takes a column, and performs the `ifelse` statement from the previous example. Once you have created the function and can see it in your environment, you can rewrite the above example like this: 
+
+```
+student_results_aggregated_suppressed <- student_results_aggregated %>%
+  mutate(g1_mean = suppress_counts(g1_mean, students),
+         g2_mean = suppress_counts(g2_mean, students),
+         g3_mean = suppress_counts(g3_mean, students),
+         students = suppress_counts(students, students))
+```
+
+However, this *still* includes some repetition we would rather avoid! therefore, we can take this one step further and make use of the `dplyr::across()` function. The `across()` function allows you to apply the same function to multiple columns - type `?dplyr::across` in the console to view the help page and more information. 
+
+We can rewrite the above in the following way:
+```
+student_results_aggregated_suppressed <- student_results_aggregated %>%
+  mutate(across(c(g1_mean, g2_mean, g3_mean, students), ~ifelse(students < 5, 'c', .)))
```

Comment:
Any reason for not using suppress_counts() instead of ifelse() in this final version?

Reply:
I couldn't get it to work! I tried

Reply:
![image](https://user-images.githubusercontent.com/101197790/233090467-4dbda11f-f6d9-40ea-a3e5-6066f9622cc2.png)
???